clang/include/clang/Basic/Attr.td

```diff
@@ -1073,6 +1073,36 @@ static llvm::StringRef canonicalizePlatformName(llvm::StringRef Platform) {
              .Case("ShaderModel", "shadermodel")
              .Default(Platform);
 }
+static std::vector<llvm::StringRef> equivalentPlatformNames(llvm::StringRef Platform) {
```

Comment:
Just return `ArrayRef` to avoid the copy?

Reply:
I tried `ArrayRef` first and it crashes. The array seems to get freed after the `llvm::find_if` returns for reasons that I don't understand.

Reply:
Ah, right, the `ArrayRef` lives but the underlying storage might not. It might be clearer if the StringSwitch is `StringSwift<std::vector<llvm::StringRef>>`.

Alternative is to initialized a static StringMap to lookup (but probably that should not live in the `.td` file).

Reply:
Oh whoops, I forgot to change that when I switched to a vector! I think it's probably best to keep this in the `.td` file